src/item_factory.cpp

```diff
@@ -2458,6 +2458,7 @@ void Item_factory::add_entry( Item_group &ig, JsonObject &obj )
                                              probability ) );
     }
     if( !sptr ) {
+        debugmsg( "Failed to load an entry in an item group." );
```

Comment:
Why not `throw` an exception? This is already done by the JSON system when the input is not valid JSON and it's already caught by the loading code. There is the handy `JsonObject::throw_error` function for that.